osm2streets/src/paint.rs

```diff
@@ -0,0 +1,235 @@
+use crate::{marking, marking::Marking};
+
+// We use geom and stay in map space. Output is done in latlon.
+use geom::{Angle, Distance, Line, PolyLine, Polygon, Pt2D, Ring};
+
+use crate::lanes::TrafficMode;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct PaintArea {
+    // Because I'm lazy and don't want to make different "map space" and "lonlat space" PaintArea
+    // types, I'm using geo::Polygon, so we can just swap out the coords in place. Not ideal.
+    /// A simple ring.
+    pub area: geo::Polygon,
```

Comment:
I don't know what the best solution is here. `InternalPaintArea` and `PaintArea` as lonlat and map space alternatives, or output types parametrised by the point type, something like:
```rs
pub struct PaintArea<P: Into<float_pair>> {
    area: Vec<P>,
    color: PaintColor,
}
```

These output types are specifically for exporting to other languages like Java, but the same level of abstraction should be useful if you are using the StreetNetwork coordinate space, like A/B Street does.

Reply:
I think using the same type for both and just documenting the difference is OK for now. If you have any ideas, we should chat with the georust folks, because they've been thinking about this too: https://docs.rs/geo/latest/geo/geometry/struct.Coord.html

> Unlike Point (which in the future may contain additional information such as an envelope, a precision model, and spatial reference system information), a Coord only contains ordinate values and accessor methods.